packages/gtmcore/gtmcore/dataset/manifest/file.py

```diff
@@ -0,0 +1,288 @@
+from typing import List, Dict, Optional, NamedTuple
+import pickle
+import os
+from enum import Enum
+from collections import OrderedDict
+import json
+import redis
+import glob
+import copy
+
+from gtmcore.dataset.dataset import Dataset
+from gtmcore.logging import LMLogger
+
+logger = LMLogger.get_logger()
+
+
+class PersistTaskType(Enum):
+    """Enumeration of persist tasks"""
+    DELETE = 0
+    ADD = 1
+    UPDATE = 2
+
+
+# NamedTuple to capture tasks to be done to persist manifest changes to disk
+PersistTask = NamedTuple('PersistTask', [('relative_path', str), ('task', PersistTaskType), ('manifest_file', str)])
+
+
+class ManfiestJSONEncoder(json.JSONEncoder):
+    """A custom json encoder to output json files that will be moderately compatible with git"""
+    def encode(self, o):
+        if isinstance(o, OrderedDict):
+            output = [f'"{x}":{json.dumps(o[x], separators=(",", ":"))}' for x in o]
```

Comment:
If you follow my suggestion to make file metadata also an ordered dict, you'll need to also specify the ManifestJSONEncoder to the json.dumps recursive-ish call here.

Also, it's probably more efficient to do:

```suggestion
            output = ",\n".join(f'"{x}":{json.dumps(o[x], separators=(",", ":"), cls=ManfiestJSONEncoder)}'
                for x in o)
```

Which won't materialize the list.

Reply:
list comprehensions are usually the fastest route, but i'll take a closer look

Reply:
Indeed, for str.join, it is faster to do it your way. There's still a redundant plus there.

https://stackoverflow.com/a/9061024

Reply:
Too much clojure, I guess.

Reply:
@davclark - just checked and your suggested code is 2x slower. List comprehensions FTW!

Going to leave this as is.